### ISSUE TITLE
Allow categorizing/labelling issues downstream

### ISSUE THREAD

**Issue by yrodiere** [hibernate/replicate-jira#44]
Currently, you can add labels to downstream copies of issues, but those labels will get erased on the next sync of the issue.

So I cannot, for instance, add a label to all issues that pertain to a given downstream concern. I would have to add the label upstream, which doesn't make much sense, and also is not very practical.

Could we find some way to prevent downstream labels from being erased on sync?

For example, we could agree that upstream labels all get prefixed with `upstream-` upon sync, that way the sync can leave all existing labels that don't start with `upstream-` as-is?

**Comment by marko-bekhta** [comment]
hmmmm 🤔 
my first thought was to leverage the `update` block of the request (https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issues/#api-rest-api-3-issue-issueidorkey-put)
```json
  "update": {
    "labels": [
      {
        "add": "triaged"
      }
    ],
```
, so that we do not make a request to GET a downstream issue. But, that way, we'd be only appending the labels and if an upstream issue removes a label, we won't reflect that ... to get that working, we'd need to fetch the downstream issue to compare... 

Let me know if we should make that extra REST call, and I'll look into it.

**Comment by yrodiere** [comment]
> Let me know if we should make that extra REST call, and I'll look into it.

I think that makes sense, and probably will be necessary for other features.

Worst case we can try to batch these extra REST calls, by always handling e.g. batches of 20 upstream issues, and fetching the corresponding 20 downstream issues before handling the batch.

**Comment by marko-bekhta** [comment]
The labels should now have the `upstream-` prefix where applicable. Fix version labels don't have the prefix as we know their format and can filter them out. 
Speaking of fix versions... I'll take a look if we can easily create and assign fix versions downstream (but maybe not today 😃)

**Comment by yrodiere** [comment]
Thanks.

> Speaking of fix versions... I'll take a look if we can easily create and assign fix versions downstream (but maybe not today 😃)

Definitely not urgent, indeed :)

**Comment by yrodiere** [comment]
@marko-bekhta I'll close this as I think it was fixed by #46
